src/shared/driverUtils.js

```diff
@@ -0,0 +1,258 @@
+    const pdServerProcess = exec(fluid.personalData.startServerCmd + " " + serverPort);
+    for (let i = 0; i < NUM_CHECK_REQUESTS; i++) {
+        try {
+            console.debug(`... attempt ${i}`);
+            resp = await axios(url);
+        }
+        catch (error) {
+            console.debug(`... attempt ${i} error: ${error.message}`);
+            resp.status = 503;
+        }
+        if (resp.status === 200) {
+            break;
+        } else {
+            console.debug(`... attempt ${i} going to sleep`);
+            await fluid.personalData.sleep(DELAY);
+        }
+    }
+    console.debug(`- Attempt to start server, result:  ${resp.status}`);
+    return { status: resp.status, process: pdServerProcess, wasRunning: false };
+};
+
+/**
+ * Stop the personal data server, if it was started by this testing sequence.
+ * If it was started in order to run the tests, the `childProcess` argument will
+ * have a non-null ChildProcess instance and a `wasRunning` flag set to false.
+ * If so, check the ChildProcess's exit status and, if it has not exited, invoke
+ * its `kill()` method.
+ *
+ * @param {Object} childProcessInfo - Information about the server's process
+ * @param {Object} childProcessInfo.wasRunning - True if the server was already
+ *                                               running.
+ * @param {Object} childProcessInfo.process - ChildProcess instance `wasRunning`
+ *                                            created if the server was not
+ *                                            running.
+ * @param {String} url - Url to use to check if the ChildProcess has been killed.
+ */
+fluid.personalData.stopServer = async function (childProcessInfo, url) {
+    console.debug(`- Stopping server at ${url}`);
+    if (childProcessInfo.wasRunning === false && childProcessInfo.process &&
+        childProcessInfo.process.exitCode === null) {
+        childProcessInfo.process.kill();
+    }
+    // Wait for the childProcess to fully exit
+    for (let i = 0; i < NUM_CHECK_REQUESTS; i++) {
+        try {
+            console.debug(`... attempt ${i}`);
+            await axios(url);
+        }
+        catch (error) {
+            break;
+        }
+        await fluid.personalData.sleep(DELAY);
+    }
+};
+
+/**
+ * Start the database using Docker.  This checks whether the docker container
+ * is already running, and if not, attempts to start it.
+ *
+ * @param {String} container - Name of the docker container.
+ * @param {String} image - Name of the associated docker image.
+ * @param {String} dbConfig - Parameters for creating a database within the
+ (*                           `container` or to check if one already exists.
+ * @param {String} dbConfig.user - Adminstrative user of the database.
+ * @param {String} dbConfig.password - Adminstrator's password.
+ * @param {String} dbConfig.port - Port for database requests.
+ * @return {Object} An object containing two flags, `wasPaused` if the container
```

Comment:
It might be more helpful to return a promise which rejects if the docker action fails (that is, to rethrow errors) - without this, users of this function tend to continue even if there has been an error